servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +468,174 @@ def test_query_jobs_submission_pagination(self):
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Create a failed job
+            job_failed = self.api_job_id(
+                self.start_job(
+                    'not_a_command',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Created a succeeded job
+            job_succeeded = self.api_job_id(
+                self.start_job(
+                    'echo SUCCEEDED',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Created a running job
+            job_running = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+
+            # Create a running job that has a different label and no test_token label
+            job_running_no_token_diff_label = self.api_job_id(
+                self.start_job('sleep 180', labels=labels_no_show, name=name))
+
+            # Create a running job that has different label values
+            job_running_diff_label_value = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=another_value,
+                    name=name,
+                    with_test_token=True))
+            # Create a running job that has different labels
+            job_running_diff_labels = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=different_labels,
+                    name=name,
+                    with_test_token=True))
+
+            # wait_status need 10ms on local provider
+            if self.provider == local.LocalJobProvider(resources):
+                sleep(10)
+
+            self.wait_status(job_aborted, ApiStatus.RUNNING)
+            self.must_abort_job(job_aborted)
+
+            self.wait_status(job_aborted, ApiStatus.ABORTED)
+
+            self.wait_status(job_failed, ApiStatus.FAILED)
+
+            self.wait_status(job_succeeded, ApiStatus.SUCCEEDED)
+
+            self.wait_status(job_running, ApiStatus.RUNNING)
+            self.wait_status(job_running_no_token, ApiStatus.RUNNING)
+            self.wait_status(job_running_no_token_diff_label,
+                             ApiStatus.RUNNING)
+            self.wait_status(job_running_diff_label_value, ApiStatus.RUNNING)
+            self.wait_status(job_running_diff_labels, ApiStatus.RUNNING)
+
+            userId = self.must_get_job(job_running).extensions.user_id
+            aggregation_resp = self.must_get_job_aggregations('HOURS_1')
+
+            self.assert_aggregation_response_equal(aggregation_resp, userId,
+                                                   name)
+
+            # Cancel jobs for testing
+            self.must_abort_job(job_running)
```

Comment:
Not sure I would assert fail here, the main goal would be to just limit runtime, so you want this to run for all jobs and you don't necessarily need the test to fail if this doesn't work. Probably I would just call abort without the assert fail (modified version of `must_abort_job`)